.changeset/curly-squids-chew.md

```diff
@@ -0,0 +1,17 @@
+---
+"@chakra-ui/pin-input": minor
+---
+
+- Move input props logic to parent hook and expose prop-getter `getInputProps`.
+  This helps to co-locate the state logic for easier debugging
+- Added support for alpha-numeric and secret values.
+- Added `type` prop which can be either `alphanumeric` or `number`
+- Added `mask` prop to hide input value similar to `<input type='password' />`
```

Comment:
Random thought on how to make something like this potentially more clear in the future:

- Added support for alpha-numeric values via new `type` prop, which can be either `alphanumeric` or `number` (default: `number`)
- Added support for secret/hidden values via the `mask` prop (default: `false`), similar to `<input type="password" />`

Just makes it a little more clear that this is 3 changes, which prop corresponds to which change, and what the defaults of these new props are. Not a huge deal but just a thought that came to mind that I wanted to share 😁

Reply:
Amazing! We can reflect this in the docs for PinInput.

In the future, feel free to update the changeset as well. I'd love that but I learned from your suggestions. Thanks for sharing!

Reply:
@segunadebayo Good idea! Didn't even think about updating them directly lol